exercises/plot_dets_filesize.py

```diff
@@ -0,0 +1,94 @@
+from databroker import Broker
+import pandas as pd
+
+import os
+
+import datetime
+import time
+from time import mktime
+
+from eiger_io.fs_handler import EigerHandler
+from databroker.assets.handlers import AreaDetectorTiffHandler
+
+
+def find_keys(hdrs, db):
+    '''
+        This function searches for keys that are stored via filestore in a
+        database, and gathers the SPEC id's from them.
+    '''
+    FILESTORE_KEY = "FILESTORE:"
+    keys_dict = dict()
+    files = []
+    for hdr in hdrs:
+        for stream_name in hdr.stream_names:
+            events = hdr.events(stream_name=stream_name)
+            events = iter(events)
+            while True:
+                try:
+                    event = next(events)
+                    if "filled" in event:
+                        # there are keys that may not be filled
+                        for key, val in event['filled'].items():
+                            if key not in keys_dict and not val:
+                                # get the datum
+                                if key in event['data']:
+                                    datum_id = event['data'][key]
+                                    resource = db.reg.resource_given_datum_id(datum_id)
+                                    resource_id = resource['uid']
+#                                    keys_dict[key] = resource['spec']
+                                    datum_gen = db.reg.datum_gen_given_resource(resource)
+                                    datum_kwargs_list = [datum['datum_kwargs'] for datum in datum_gen]
+                                    fh = db.reg.get_spec_handler(resource_id)
+                                    file_lists = fh.get_file_list(datum_kwargs_list)
+                                    file_sizes = get_file_size(file_lists)
+                                    files.append(file_sizes)
+                                    if key not in keys_dict:
```

Comment:
if you use `defaultdict` (see above), all you need is:
`keys_dict[key] += file_sizes`

Reply:
Wow, that is a nice trick, thanks!